logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/Conversation.kt

```diff
@@ -5,13 +5,42 @@ import com.wire.kalium.logic.data.id.QualifiedID
 import com.wire.kalium.logic.data.id.TeamId
 import com.wire.kalium.logic.data.user.User
 import com.wire.kalium.logic.data.user.UserId
-
 typealias ConversationId = QualifiedID
 
 data class Conversation(val id: ConversationId, val name: String?, val type: Type, val teamId: TeamId?) {
     enum class Type { SELF, ONE_ON_ONE, GROUP }
 }
 
+sealed class ConversationDetails(val conversation: Conversation) {
+
+    class Self(conversation: Conversation) : ConversationDetails(conversation)
+
+    class OneOne(
+        conversation: Conversation,
+        val contactName: String,
+        val connectionState: ConnectionState,
+        val federationStatus: FederationStatus,
+        val legalHoldStatus: LegalHoldStatus
+    ) : ConversationDetails(conversation) {
+        enum class ConnectionState {
+            // The other user has sent a connection request to this one
+            INCOMING,
+
+            // This user has sent a connection request to another user
+            OUTGOING,
+
+            // The connection is complete and the conversation is in its normal state
+            ACCEPTED
+        }
+    }
+
+    class Group(conversation: Conversation) : ConversationDetails(conversation)
```

Comment:
Currently there's no `FederationStatus` or `LegalHoldStatus` for `Group` conversations. But I'm sure at least legal hold can show up here in the not-so-near future.